tests/cases/compiler/expandoFunctionNullishProperty.ts

```diff
@@ -0,0 +1,27 @@
+// @strict: true
+// @target: esnext
+// @declaration: true
+
+// mentioned in https://github.com/microsoft/TypeScript/issues/54220
+
+interface TestNull {
+  (): void;
+  readonly prop: null;
+}
+
+export function testNull(): TestNull {
+  function inner() {}
+  inner.prop = null;
+  return inner;
```

Comment:
please add a test that shows `inner` is assignable to a more realistic type like

```ts
interface TestStringNull {
  (): void;
  prop: string | null
}
```

Since `readonly prop: null` is intentionally useless (and why the JS rule existed in the first place).

Reply:
added